crates/innertx/src/cache_utils.rs

```diff
@@ -0,0 +1,65 @@
+use crate::innertx_inspector::InternalTransaction;
+use alloy_primitives::{BlockHash, TxHash, B256};
+use eyre::Report;
+use moka::sync::Cache;
+use once_cell::sync::OnceCell;
+
+// Cache for BlockTable: block_hash -> Vec<TxHash>
+static BLOCK_CACHE: OnceCell<Cache<B256, Vec<TxHash>>> = OnceCell::new();
+
+// Cache for TxTable: tx_hash -> Vec<InternalTransaction>
+static TX_CACHE: OnceCell<Cache<B256, Vec<InternalTransaction>>> = OnceCell::new();
```

Comment:
I was actually thinking of this, since the types themselves should be hashable and the names give meaning to what the keys actually represent. In doing so, we can also drop the comments since the code becomes self-explanatory.
```suggestion
// Cache for BlockTable: block_hash -> Vec<TxHash>
static BLOCK_CACHE: OnceCell<Cache<BlockHash, Vec<TxHash>>> = OnceCell::new();

// Cache for TxTable: tx_hash -> Vec<InternalTransaction>
static TX_CACHE: OnceCell<Cache<TxHash, Vec<InternalTransaction>>> = OnceCell::new();
```